core/src/test/java/org/infinispan/stream/BaseSetupStreamIteratorTest.java

```diff
@@ -48,9 +49,12 @@ protected void enhanceConfiguration(ConfigurationBuilder builder) {
    @Override
    protected void createCacheManagers() throws Throwable {
       builderUsed = new ConfigurationBuilder();
-      BaseControlledConsistentHashFactory<? extends ConsistentHash> chf =
-            cacheMode.isScattered() ? new TestScatteredConsistentHashFactory() : new TestDefaultConsistentHashFactory();
-      builderUsed.clustering().cacheMode(cacheMode).hash().numSegments(3).consistentHashFactory(chf);
+      HashConfigurationBuilder hashConfiguration = builderUsed.clustering().cacheMode(cacheMode).hash().numSegments(3);
+      if (!cacheMode.isReplicated()) {
```

Comment:
I don't like the idea of having a random CH in the repl test, but if it works I guess it's fine.

Reply:
The problem is that the test assigns owners where not all are owned in a REPL cache. So it ends  up being the same issue as wait for initial state transfer.

Reply:
Sure, I didn't mean we should use the existing `TestDefaultConsistentHashFactory`. Looking back at [ISPN-7997](https://issues.jboss.org/browse/ISPN-7997) the default `SyncReplConsistentHashFactory` will allocate one node to be primary owner of each segment so it should be good enough, I would have just preferred a more deterministic implementation similar to `TestDefaultConsistentHashFactory`.